### ISSUE TITLE
How can I use it as a global module?

### ISSUE THREAD

**Issue by alex4506** [BJS-kr/nestjs-omacache#35]

**Comment by BJS-kr** [comment]
sorry for late. I'll add global option as soon as possible

**Comment by AnantaPodder** [comment]
@BJS-kr Can I work on this issue?

**Comment by BJS-kr** [comment]
@AnantaPodder of course!

**Comment by AnantaPodder** [comment]
@BJS-kr Do user always have to create custom cache Decorator with storage? If yes, User can define property descriptor and use it anywhere in the code without need of global module.

**Comment by BJS-kr** [comment]
Yes. what this package provides is cache decorator factory, not decorator itself. And, yes, I intended global usage of user defined decorators. It's even possible to use without importing module. CacheModule is needed only for set-on-start caching.  I thought @alex4506 asking about more... 'NestJS way' module. Like, work with @Global() decorator, and configurable styles.

**Comment by AnantaPodder** [comment]
As I see, we are discovering all controller and providers using `ServiceDiscovery` in [getAllInstances()](https://github.com/BJS-kr/nestjs-omacache/blob/21eed8497e2f75cd12c6d335b6824d10aabc2dc5/lib/cache.service.ts#L16). So I think it's already behaving like a global module.

**Comment by BJS-kr** [comment]
@AnantaPodder yes. So the issue is already been resolved actually😅. sorry for misunderstanding.
The reason I didn't closed this issue is, I was planning to change cache module not to be global by default. I think it is not idiomatic way for NestJS. So, after changes applied, discovery service should be modified for dynamic module scopes include @Global().  I had to tell you this first. sorry
